### ISSUE TITLE
Name plates still show even if player has me blocked

### ISSUE THREAD

**Issue by epicEaston197** [Yellow-Dog-Man/Resonite-Issues#1385]
### Describe the bug?

I could still see a players name plate if they have blocked me

### To Reproduce

Have three people in a world the host is just there to facilitate the testing

!. Have one user block another user

2. You will observe that the nameplates are still visible

I've seen this happen to default nameplates but custom ones as well

### Expected behavior

When a user blocks me I should not be able to see their name played and they should not be able to see my name played either

### Screenshots

No screenshots because I don't wanna call anyone out

### Resonite Version Number

2024.2.12.1430

### What Platforms does this occur on?

Windows

### What headset if any do you use?

desktop

### Log Files

I don't think log files are necessary but will provide if requested

### Reporters

@epicEaston197

**Comment by JackTheFoxOtter** [comment]
Well, you'd still want to see SOMETHING though. Making the user disappear entirely can lead to a bunch of awkward social situations where two people that have one another blocked are standing in the same place, trying to talk to the same person without knowing someone else is already doing so.

A elegant solution I believe would be to display blocked users as faceless "fallback" avatars, like, just silhouettes of a person, that perhaps even glows up if they speak, and make a subtle sound effect so you know they are speaking, even if you can't hear them.

I definitely think custom nameplates of a blocked user should be replaced with the default nameplate at the very least though. Perhaps even display the name on it as "Blocked User" instead of the username and hide the badges. But until we have a better visualization of blocked users, there probably should be at least something visible for you to be aware that there is a person that you can't see, but others likely can.

**Comment by shiftyscales** [comment]
@epicEaston197 @JackTheFoxOtter - can you confirm/provide screenshots that show a users' custom nameplate not being hidden by the blocking functionality? As noted- at the least, it would be good to default to the default nameplate as all renderers under a blocked user should ideally be hidden by default.

Additional customizability of how blocking functions, e.g. whether or not to use placeholders could possibly come alongside further extensions to the blocking functionality- the current implementation was just a primitive one because it's better to have the functionality at all than not have it, even if it is simple in its current design.

**Comment by epicEaston197** [comment]
> @epicEaston197 @JackTheFoxOtter - can you confirm/provide screenshots that show a users' custom nameplate not being hidden by the blocking functionality?

Yes I can confirm this it does happen to me and yes I'll get a screenshot the next opportunity I can

**Comment by epicEaston197** [comment]
@shiftyscales a log file was provided in issue #2097

**Comment by shiftyscales** [comment]
Additional report by @SlyTheFloof in #2097:

> ### Describe the bug?
> 
> Can see mutual blocked users name tags
> 
> ### To Reproduce
> 
> idk 
> 
> ### Expected behavior
> 
> not to see anyone
> 
> ### Screenshots
> 
> ![image](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/56106431/96900fac-44da-4257-bb5b-9cba034979ee)
> 
> 
> ### Resonite Version Number
> 
> Beta 2024.5.24.202
> 
> ### What Platforms does this occur on?
> 
> Windows
> 
> ### What headset if any do you use?
> 
> Desktop, Steam VR, Oculus
> 
> ### Log Files
> 
> [Uploading SLYS-DESKTOP-PC - 2024.5.24.202 - 2024-05-24 20_11_59.log…]()
> 
> .
> 
> ### Additional Context
> 
> _No response_
> 
> ### Reporters
> 
> _No response_

**Comment by epicEaston197** [comment]
here's my own screenshot
![image](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/76523692/4a46ce6a-f5b2-4dad-8b93-57aa0e0fe4b9)

**Comment by SlyTheFloof** [comment]
![image](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/56106431/90e7d362-32d2-49c4-a8fe-4a5b17a70714)

**Comment by SlyTheFloof** [comment]
![image](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/56106431/54bc75f0-8b8e-4007-a0cb-024202e8a6b0)
more issue with custom name badges showing blocked users
